source/utilities.cc

```diff
@@ -938,12 +948,166 @@ namespace aspect
                                      const MPI_Comm &comm)
     {
       std::string data_string;
+      std::stringstream urlString;
 
       if (Utilities::MPI::this_mpi_process(comm) == 0)
         {
           // set file size to an invalid size (signaling an error if we can not read it)
           unsigned int filesize = numbers::invalid_unsigned_int;
 
+          /**
+           * Lookup and use data from the provided url
+           * Author: Kodi Neumiller
+           * Added: 10/9/18
+           **/
+          //----Only run if the user wishes to use the libdap packages----//
+#ifdef HAVE_LIBDAP
+          //Check to see if the prm file will be reading data from the disk or
+          // from a provided URL
+          if (filename.find("http://") == 0 || filename.find("https://") == 0 || filename.find("file://") == 0)
+            {
+              libdap::Connect *url = 0;
+              url = new libdap::Connect(filename);
```

Comment:
1. Move declaration and assignment into a single line.
2. You never seem to delete this object again, meaning you are leaking memory. Why do you not do ``libdap::Connect url(filename);``?

Reply:
Fixed, thanks for pointing that out.